wire-library/wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/GrpcCalls.kt

```diff
@@ -87,3 +97,118 @@ fun <S : Any, R : Any> GrpcCall(function: (S) -> R): GrpcCall<S, R> {
     override fun clone() = GrpcCall(function)
   }
 }
+
+/**
+ * Returns a new instance of [GrpcStreamingCall] that can be used for a single call to
+ * [execute][GrpcStreamingCall.execute] or [executeBlocking][GrpcStreamingCall.executeBlocking].
+ *
+ * The returned instance launches [function] on [Dispatchers.IO]. The function must close the
+ * [SendChannel] when it has no more messages to transmit. If [function] throws, both channels will
+ * be closed using the thrown exception as a cause.
+ *
+ * This method is useful when implementing the interfaces that are generated by Wire:
+ *
+ * ```
+ *   override fun RouteChat(): GrpcStreamingCall<RouteNote, RouteNote> {
+ *     return GrpcStreamingCall { requests, responses ->
+ *       requests.consumeEach { note ->
+ *         responses.send(translateNote(note))
+ *       }
+ *       responses.close()
+ *     }
+ *   }
+ * ```
+ *
+ * It is succinct when used in an expression function:
+ *
+ * ```
+ *  override fun RouteChat() = GrpcStreamingCall<RouteNote, RouteNote> { requests, responses ->
+ *    requests.consumeEach { note ->
+ *      responses.send(translateNote(note))
+ *    }
+ *    responses.close()
+ *  }
+ * ```
+ */
+@JvmName("grpcStreamingCall")
```

Comment:
This one’s a lot more difficult to implement from Java because it uses coroutines.